file_loader.py

```diff
@@ -0,0 +1,328 @@
+"""
+Convert DICOM images to NIfTI format and identify images for the further analysis.
+
+Namely, the script:
+    - run dcm2niix command to convert DICOM images to NIfTI format
+    - prompts the user to select the images for further processing
+    - validates file existence
+    - checks for .bval and .bvec files for DWI image
+    - provides information about the images' dimensions and pixel sizes
+
+Requirements:
+    - dcm2niix -- see the Installation section in the README.md file
+
+Input file structure:
+
+    └── source_data
+        └── dir_20231010
+            ├── MRc.1.3.12.2.543543
+            ├── ...
+            └── SRe. 1.3.12.2.5432233
+
+Output file structure:
+
+    ├── bids
+    │   └── sub-001
+    │       └── ses-01
+    │           ├── anat
+    │           │   ├── sub-001_ses-01_T2w.json
+    │           │   └── sub-001_ses-01_T2w.nii.gz
+    │           └── dwi
+    │               ├── sub-001_ses-01_dwi.bval
+    │               ├── sub-001_ses-01_dwi.bvec
+    │               ├── sub-001_ses-01_dwi.json
+    │               └── sub-001_ses-01_dwi.nii.gz
+    └── source_data
+        └── dir_20231010
+            ├── MRc.1.3.12.2.543543
+            ├── ...
+            └── SRe. 1.3.12.2.5432233
+
+Author: Jan Valosek and Claude 3.5 Sonnet
+"""
+
+# TODO: write a new entry to the participant.tsv file based on the provided information (add sex and age args?)
+# TODO: use logging instead of print statements
+
+
+import os
+import shutil
+import argparse
+import pandas as pd
+import nibabel as nib
+
+
+def get_parser():
+    """
+    Parse command-line arguments.
+
+    Returns:
+    argparse.Namespace: Parsed arguments
+    """
+    parser = argparse.ArgumentParser(
+        description="Convert DICOM to NIfTI and identify images for the further analysis."
+    )
+    parser.add_argument(
+        "-dicom-folder",
+        help="Path to the folder containing DICOM images. "
+             "Example: ~/sci-balgrist-study/sourcedata/dir_20230711",
+        required=True
+    )
+    parser.add_argument(
+        "-bids-folder",
+        help="Path to the BIDS folder where the converted NIfTI images will be stored. "
+             "Example: ~/sci-balgrist-study/bids",
+        required=True
+    )
+    parser.add_argument(
+        "-participant",
+        help="Participant ID. Example: sub-001",
+        required=True
+    )
+    parser.add_argument(
+        "-session",
+        help="Session ID. Example: ses-01",
+        required=True
+    )
+    parser.add_argument(
+        "-contrasts",
+        help="MRI contrasts to use. Example: T2w dwi",
+        nargs='+',
+        default=["T2w", "dwi"],
+        type=list,
```

Comment:
Hmm, the formating of contrast args is weird, I entered `T2w`

`Please specify the row number of the ['T', '2', 'w'] image you want to use:`